libraries/AP_Networking/AP_Networking_PPP.cpp

```diff
@@ -169,28 +174,33 @@ void AP_Networking_PPP::ppp_loop(void)
     while (!hal.scheduler->is_system_initialized()) {
         hal.scheduler->delay_microseconds(1000);
     }
-    
-    // connect and set as default route
-    LWIP_TCPIP_LOCK();
-    ppp_connect(ppp, 0);
-
-    netif_set_default(pppif);
-    LWIP_TCPIP_UNLOCK();
 
     // ensure this thread owns the uart
     uart->begin(0);
     uart->set_unbuffered_writes(true);
 
     while (true) {
-        uint8_t buf[1024];
-        auto n = uart->read(buf, sizeof(buf));
-        if (n > 0) {
-            LWIP_TCPIP_LOCK();
-            pppos_input(ppp, buf, n);
-            LWIP_TCPIP_UNLOCK();
-        } else {
-            hal.scheduler->delay_microseconds(200);
+        // connect and set as default route
+        LWIP_TCPIP_LOCK();
+        ppp_connect(ppp, 0);
+
+        netif_set_default(pppif);
+        LWIP_TCPIP_UNLOCK();
+
+        need_restart = false;
+
+        while (!need_restart) {
+            uint8_t buf[1024];
+            auto n = uart->read(buf, sizeof(buf));
+            if (n > 0) {
+                LWIP_TCPIP_LOCK();
+                pppos_input(ppp, buf, n);
+                LWIP_TCPIP_UNLOCK();
+            } else {
+                hal.scheduler->delay_microseconds(200);
+            }
         }
+        GCS_SEND_TEXT(MAV_SEVERITY_INFO, "PPP: restarting");
```

Comment:
Is this GCS msg necessary?

Reply:
I agree - when I saw it in the messages, it didn't help me with anything at all, however just before this you see something like PPP timeout - which also seems unnecessary.

Reply:
I agree about changing the other messages too. Error messages, sure, but normal behavior shouldn't need to spam the gcs

Reply:
I've changed it to "PPP: connected"
none of this is "normal behaviour"